### PR TITLE
Allow github builders to build pull requests on demand

### PR DESCRIPTION
**Pull request by pablogsal**
Allow buildbots to test a pull request and report to it if the PR contains the label ":hammer: test-with-buildbots"

### CODE REVIEW

**Comment by pablogsal** [comment]
Check out https://github.com/python/cpython/pull/17816 as a test.

The label is `:hammer: test-with-buildbots`

**Comment by pablogsal** [comment]
I am super excited about this change :tada: :blush:

File: master/master.cfg [hunk 1]
```diff
@@ -430,6 +435,67 @@ for branch_num, (git_url, branchname, git_branch) in enumerate(git_branches):
...
+pull_request_builders = []
+
+for name, worker, buildfactory, stability in builders:
+    if stability != STABLE:
+        continue
+
+    if "Windows XP" in name or "VS9.0" in name:
+        continue
+
+    buildername = name + " " + "PullRequest"
```
Comment (zware) [inline master/master.cfg]: Just to bikeshed, I think I'd prefer just "PR".

**Comment by pablogsal** [comment]
> I don't pretend to fully understand every line of this, but I gather that a good chunk of it is copied from GitHubEventHandler?

The wonders of subclassing to change a bunch of  lines of a method. 😅

Will update the PR soon with your feedback :)

File: master/master.cfg (hunk 1 above)
Comment (pablogsal) [reply master/master.cfg]: Amended in 222b16f

File: master/custom/pr_testing.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+def should_pr_be_tested(change):
+    if TESTING_LABEL in {
+        label["name"]
+        for label in change.properties.asDict().get("github.labels", [set()])[0]
+    }:
+        print(f"Label detected in PR {change.branch} (commit {change.revision})")
+        return True
+    print(f"Label not found in PR {change.branch} (commit {change.revision})")
```
[the remark this answers is not in the document]
Comment (pablogsal) [reply master/custom/pr_testing.py]: Amended in 222b16f

**Pull request merged**